code/modules/sanity/sanity_mob.dm

```diff
@@ -158,6 +158,9 @@ GLOBAL_VAR_INIT(GLOBAL_INSIGHT_MOD, 1)
 		return
 	var/vig = owner.stats.getStat(STAT_VIG)
 	for(var/atom/A in view(owner.client ? owner.client : owner))
+		var/mob/M = A
+		if(ismob(M) && M?.faction == owner.faction)
+			return
 		if(A.sanity_damage) //If this thing is not nice to behold
 			. += SANITY_DAMAGE_VIEW(A.sanity_damage, vig, get_dist(owner, A))
 
```

Comment:
this meaks the idealist perk no longer work

Reply:
Then we have idealist if()'d first, then handle if we take sanity damage from mobs